### PR TITLE
Issue 68 and 102

### LINKED ISSUE #68: [New Feature]: On-demand daac data subscriber processing
### Checked for duplicates

Yes - I've already checked

### Alternatives considered

No - I haven't considered

### Related problems

_No response_

### Describe the feature request

PCM should use Tosca UI to submit a job to query and download products from DAACs by entering geographic information and/or time range. 

Following fields are needed to fill
- provider: LP DAAC(LP CLOUD), ASF DAAC, PO DAAC (PO CLOUD?)
- start datetime
- end datetime
- bounding box: optional
- ISL bucket name

### PR DESCRIPTION
**Pull request by maseca**
Closes #68 and #102
See also: https://github.com/nasa/opera-sds-lambdas/pull/7

To test, use hysds_ui on demand job feature in browser. There are query jobs specifying datetime ranges or elapsed minutes for L30 and S30. There is a single download job for all HLS products.

I also included a bugfix for the granule processing which was the result of a typo.

### CODE REVIEW

File: data_subscriber/hls/hls_catalog.py [hunk 1]
```diff
@@ -41,8 +41,11 @@ def delete_index(self):
 
     def get_all_undownloaded(self):
         undownloaded = self._query_undownloaded()
-        return [{"s3_url": result['_source']['s3_url'], "https_url": result['_source']['https_url']}
-                for result in undownloaded]
+        if undownloaded:
+            return [{"s3_url": result['_source']['s3_url'], "https_url": result['_source']['https_url']}
+                    for result in undownloaded]
+        else:
```
Comment (chrisjrd) [inline data_subscriber/hls/hls_catalog.py:47]: unnecessary `else`. the list comprehension should be able to return an empty list `[]` which is preferred over `None`.

File: docker/hysds-io.json.hlsl30_query_minutes [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      "name": "bounding_box",
+      "from": "submitter",
+      "placeholder": "-180,-90,180,90",
+      "optional": true
+    },
+    {
+      "name": "provider",
+      "from": "submitter",
+      "placeholder": "LPCLOUD",
+      "optional": true
```
Comment (chrisjrd) [inline docker/hysds-io.json.hlsl30_query_minutes:21]: is this really optional?

Comment (hhlee445) [reply docker/hysds-io.json.hlsl30_query_minutes:21]: provider should be required

Comment (maseca) [reply docker/hysds-io.json.hlsl30_query_minutes:21]: Provider defaults to LPCLOUD in the script. I can change the logic

**Comment by maseca** [comment]
Resolving the requested changes required changes be made to the lambda repo: https://github.com/nasa/opera-sds-lambdas/pull/7

**Pull request merged**